packages/grafana-e2e/src/support/types.ts

```diff
@@ -62,3 +66,11 @@ export const pageFactory = <S extends Selectors>({ url, selectors }: PageFactory
     selectors,
   };
 };
+
+type Component<S> = Omit<Page<S>, 'visit'>;
+type ComponentFactoryArgs<S> = Omit<PageFactoryArgs<S>, 'url'>;
+
+export const componentFactory = <S extends Selectors>(args: ComponentFactoryArgs<S>): Component<S> => {
+  const { visit, ...rest } = pageFactory(args);
+  return rest;
+};
```

Comment:
I like this work. This is on the same line with my thoughts, as a User of this small framework I should also be able to add Objects to a PageFactory directly not just Selectors. The result could be a components property but how to concatenate the selectors from both Selectors and Components in the PageFactory I'm not sure how to do in this moment. WDYTH @s-h-a-d-o-w 

We should do this in a separate PR.

```ts
import {BackButton} from '../../backButton';

const somePage = pageFactory({
  url: 'some url',
  selectors: {
    btn: 'Some button',
  },
  objects: {
    backButton: BackButton,
  }
});

....
aria-label={e2e.Pages.somePage.selectors.btn}
...
aria-label={e2e.Pages.somePage.selectors.backButton}
```